src/utils/ProfileProvider.tsx

```diff
@@ -58,31 +58,16 @@ export default function ProfileProvider({ children }: { children: ReactNode }) {
   const loadProfile = useCallback(async () => {
     setProfileReady(false);
 
-    const { data: sessionData, error } = await supabase.auth.getSession();
-
-    if (error) throw error;
-    if (
-      !sessionData ||
-      !sessionData.session ||
-      !sessionData.session.user ||
-      !sessionData.session.user.id
-    )
-      return;
-
-    const sessionUserEmail = sessionData.session.user.email;
-    setUserEmail(sessionUserEmail);
-
-    const sessionUserId = sessionData.session.user.id as UUID;
-    setUserId(sessionUserId);
+    if (!auth?.userId) return;
 
     await Promise.all([
-      fetchProfileById(sessionUserId).then(data => setProfileData(data)),
-      fetchLanguagesById(sessionUserId).then(data => setProfileLangs(data)),
-      fetchRolesById(sessionUserId).then(data => setProfileRoles(data)),
+      fetchProfileById(auth?.userId).then(data => setProfileData(data)),
+      fetchLanguagesById(auth?.userId).then(data => setProfileLangs(data)),
+      fetchRolesById(auth?.userId).then(data => setProfileRoles(data)),
```

Comment:
nit: correct me if I'm wrong, but since we check for `auth?.userId` (which checks `auth` and `auth.userId`), I think we can just drop the question marks here